jennchenn/uwexplore
# Populate db with CEAB requirements

## Implementation description

- wrote script to seed courses in db with CEAB values
- wrote script to seed requirements into `requirements` collection 
- updated return of `/ceab` route
- NOTE: some code change overlap with #66, but otherwise pretty standalone



## Steps to test

1. Check that you can see CEAB values for courses now
```
curl "localhost:5000/courses?query=ece406"
```
2. Login as user
```
curl -X POST -H "Content-Type: application/json" -d '{"email": "breadfydp+egg@gmail.com", "password": "<type password here>"}' localhost:5000/auth/login
```
3. Try to fetch CEAB requirements by user
```
curl -H "Authorization: Bearer <id token>" localhost:5000/ceab
```
4. Try to fetch CEAB requirements by schedule
```
curl "localhost:5000/ceab?schedule_id=64127ede93deee8bdc7a9121"
```



## What should reviewers focus on?

- return object looks like:
```
{
  "CSE ALL": {
    "completed": 0,
    "requirement": 6
  },
  "CSE WEIGHT": {
    "completed": 197.0,
    "requirement": 225
  },
  "ENG DES": {
    "completed": 54.0,
    "requirement": 225
  },
  "ENG SCI": {
    "completed": 18.0,
    "requirement": 225
  },
  "ENG SCI & ENG DES": {
    "completed": 72.0,
    "requirement": 900
  },
  "LIST A": {
    "completed": 0,
    "requirement": 1
  },
  "LIST B": {
    "completed": 0,
    "requirement": 1
  },
  "LIST C": {
    "completed": 2,
    "requirement": 2
  },
  "LIST D": {
    "completed": 3,
    "requirement": "NA"
  },
  "MATH": {
    "completed": 42.0,
    "requirement": 195
  },
  "MATH & SCI": {
    "completed": 42.0,
    "requirement": 420
  },
  "PD COMP": {
    "completed": 0,
    "requirement": 2
  },
  "PD ELEC": {
    "completed": 1,
    "requirement": 3
  },
  "SCI": {
    "completed": 0.0,
    "requirement": 195
  },
  "TE": {
    "completed": 2,
    "requirement": 6
  },
  "TE & CSE": {
    "completed": 2,
    "requirement": 13
  }
}
```
- currently seeding db based on CSV dumps of the tables in the db
- will need to do some manual checking later that these values match the ones in current planner
- calculations are done on request rn (i.e. no saving of completed totals per user atm)
